vsphere/datadog_checks/vsphere/config.py

```diff
@@ -178,10 +175,16 @@ def _parse_resource_filters(self, all_resource_filters):
                     resource_filter['type'],
                     ALLOWED_FILTER_TYPES,
                 )
-            is_whitelist = resource_filter['type'] == 'whitelist'
-
-            filter_key = ResoureFilterKey(resource_filter['resource'], resource_filter['property'], is_whitelist)
-            if filter_key in formatted_resource_filters:
+            patterns = [re.compile(r) for r in resource_filter['patterns']]
+            FilterClass = FILTER_PROP_TO_FILTER_CLASS[resource_filter['property']]
+
+            filter_instance = FilterClass(
+                resource_filter['resource'],
+                resource_filter['property'],
+                patterns,
+                is_whitelist=(resource_filter['type'] == 'whitelist'),
+            )
```

Comment:
It might be more natural to provide a "strategy" constructor like this in resource_filters.py:

```python
def create_resource_filter(property, resource_type, property_name, patterns, is_whitelist=True):
            FilterClass = FILTER_PROP_TO_FILTER_CLASS[property]
            filter_instance = FilterClass(
            ...
            )
            return filter_instance
```

That way we encapsulate most of the logic in resource_filters.py.

Reply:
Good idea 👍